src/Sarif/ExtensionMethods.cs

```diff
@@ -253,86 +254,26 @@ public static string GetFirstSentence(string text)
                 return text;
             }
 
-            int length = 0;
-            bool withinQuotes = false;
-            bool withinParentheses = false;
-            bool lastEncounteredWasDot = false;
-            bool withinEllipsis = false;
+            // We will return at most the first line
+            string[] lineBreaks = { Environment.NewLine, "\r", "\n" };
 
-            foreach (char ch in text)
+            foreach (string s in lineBreaks)
             {
-                length++;
-                switch (ch)
-                {
-                    case '\'':
-                    {
-                        // we'll ignore everything within parenthized text
-                        if (!withinParentheses)
-                        {
-                            withinQuotes = !withinQuotes;
-                        }
-                        lastEncounteredWasDot = false;
-                        break;
-                    }
+                int index = text.IndexOf(s);
 
-                    case '(':
-                    {
-                        if (!withinQuotes)
-                        {
-                            withinParentheses = true;
-                        }
-                        lastEncounteredWasDot = false;
-                        break;
-                    }
-
-                    case ')':
-                    {
-                        if (!withinQuotes)
-                        {
-                            withinParentheses = false;
-                        }
-                        lastEncounteredWasDot = false;
-                        break;
-                    }
-
-                    case '.':
-                    {
-                        if (withinQuotes || withinParentheses || withinEllipsis) { continue; }
-                        if (length < text.Length && text[length] == '.')
-                        {
-                            withinEllipsis = true;
-                            lastEncounteredWasDot = false;
-                            break;
-                        }
-
-                        lastEncounteredWasDot = true;
-                        break;
-                    }
-
-                    // If we encounter a line-break, we return all leading text.
-                    case '\n':
-                    case '\r':
-                    {
-                        if (withinQuotes || withinParentheses) { continue; }
-                        return text.Substring(0, length).TrimEnd('\r', '\n', ' ', '.') + ".";
-                    }
+                if (index > -1)
+                {
+                    text = text.Substring(0, index);
```

Comment:
>text [](start = 20, length = 4)

Once you've found the first line break character, you won't find any more, so `break` here.

Reply:
Good eye

---
In reply to: [213073698](https://github.com/Microsoft/sarif-sdk/pull/1050#discussion_r213073698) [](ancestors = 213073698)